### PR TITLE
docs: fix multiple broken links and stale type interface

### LINKED ISSUE #424: docs: multiple broken links
https://github.com/textlint/textlint/blob/master/docs/formatter.md

> Pass following array of [TextLintResult](https://github.com/textlint/textlint/blob/master/typings/textlint.d.ts "TextLintResult") to reporter module.
> `TextLintMessage` and `TextLintResult` are defined in [textlint.d.ts](https://github.com/textlint/textlint/blob/master/typings/textlint.d.ts "textlint.d.ts").

Edit: add tasks (@0x6b)

- [x] [Advanced: Paragraph rule](https://github.com/textlint/textlint/blob/master/docs/rule-advanced.md)
- [x] ~~[Configuring textlint](https://github.com/textlint/textlint/blob/master/docs/configuring.md)~~ (no 404 or 301)
- [x] [Creating Rules](https://github.com/textlint/textlint/blob/master/docs/rule.md)
- [x] [Filter rule](https://github.com/textlint/textlint/blob/master/docs/filter-rule.md)
- [x] [Formatter](https://github.com/textlint/textlint/blob/master/docs/formatter.md)
- [x] [Getting Started with textlint](https://github.com/textlint/textlint/blob/master/docs/getting-started.md)
- [x] [Plugin](https://github.com/textlint/textlint/blob/master/docs/plugin.md)
- [x] [TxtAST interface](https://github.com/textlint/textlint/blob/master/docs/txtnode.md)
- [x] [Use as node modules](https://github.com/textlint/textlint/blob/master/docs/use-as-modules.md)

### PR DESCRIPTION
**Pull request by 0x6b**
- [Advanced: Paragraph rule](https://github.com/textlint/textlint/blob/master/docs/rule-advanced.md)
- [Creating Rules](https://github.com/textlint/textlint/blob/master/docs/rule.md)
- [Filter rule](https://github.com/textlint/textlint/blob/master/docs/filter-rule.md)
- [Formatter](https://github.com/textlint/textlint/blob/master/docs/formatter.md)
- [Getting Started with textlint](https://github.com/textlint/textlint/blob/master/docs/getting-started.md)
- [Plugin](https://github.com/textlint/textlint/blob/master/docs/plugin.md)
- [TxtAST interface](https://github.com/textlint/textlint/blob/master/docs/txtnode.md)
- [Use as node modules](https://github.com/textlint/textlint/blob/master/docs/use-as-modules.md)

Closes #424. As discussed at #434, documentation could be improved but focused dead link and stale type declaration at this moment.

### CODE REVIEW

**Comment by 0x6b** [comment]
I forget to run test locally before push as only documentation change. Reviewing.

File: docs/txtnode.md [hunk 1]
```diff
@@ -150,7 +150,7 @@ Following parsers are built-in.
 | [`markdown-to-ast`](../packages/markdown-to-ast) | [![npm](https://img.shields.io/npm/v/markdown-to-ast.svg?style=flat-square)](https://www.npmjs.com/package/markdown-to-ast) | markdown parser |
 | [`txt-to-ast`](../packages/txt-to-ast) | [![npm](https://img.shields.io/npm/v/txt-to-ast.svg?style=flat-square)](https://www.npmjs.com/package/txt-to-ast) | plain text parser |
 
-If you want to get other type, please [create new issue](https://github.com/textlint/textlint/issues/new).
+If you want to get other type, please [create new issue](https://github.com/login?return_to=https%3A%2F%2Fgithub.com%2Ftextlint%2Ftextlint%2Fissues%2Fnew).
```
Comment (azu) [inline docs/txtnode.md]: `https://github.com/textlint/textlint/issues/new` should be ignored

Comment (0x6b) [reply docs/txtnode.md]: Thank you. Fixed at bd2089a.

**Comment by 0x6b** [comment]
Merged as CI passed successfully and got LGTM from @azu. For the record use following configuration while editing:

```json
{
  "rules": {
    "no-dead-link": {
      "checkRelative": true,
      "ignore": [
        "https://www.npmjs.com/search?q=textlint",
        "https://github.com/textlint/textlint/issues/new"
      ]
    }
  }
}
```

**Pull request merged**